src/backend/distributed/transaction/transaction_management.c

```diff
@@ -353,6 +354,14 @@ CoordinatedTransactionCallback(XactEvent event, void *arg)
 
 			ResetGlobalVariables();
 
+			/*
+			 * Clear MetadataCache table if we're aborting from a CREATE EXTENSION Citus
+			 */
+			if (IsCreatingCitus())
+			{
+				InvalidateMetadataSystemCache();
+			}
```

Comment:
@marcocitus I'm not very familiar with the workflow but I was thinking of setting the creatingCitus flag back to false after InvalidatingMetadataSystemCache(). I think sub_aborts would happen before xact_aborts so that shouldn't be a problem but I wanted to double check if that understanding is correct?

Reply:
I think we should reset the flag in both cases, since they can happen somewhat independently from each other.

Reply:
What do you mean "both cases"? commit/abort or main/subxact?

We should set it to false on transaction rollback or commit.

But for subabourt/subcommit, it's a little trickier. To get that right, maybe we need to make the variable a (sub)transaction ID rather than a boolean. Instead of setting it to true, we set it to the current (sub)transaction ID. And on commit/rollback, we check if the variable matches the (sub)transaction that's ending, and if so, we invalidate it if it's a rollback, and set the variable back to `InvalidTransactionId`.